services/roomService/src/lib/CoveyTownController.ts

```diff
@@ -154,18 +165,48 @@ export default class CoveyTownController {
       return d <= Math.max(0, incomingMessage.broadcastRadius);
     };
     const nearbyPlayerSessions = this._sessions.filter(s => isInChatRadius(s.player));
-    const receivingPlayers = nearbyPlayerSessions.map(s => s.player).filter(p => p.id !== sender.id);
+
+    const notBlocked = (s: PlayerSession) => {
+      const length = this._blockedPlayerSessions.get(session)?.length;
+      for(let i=0; i<(length as number); i++) {
+        if (this._blockedPlayerSessions.get(session)?.includes(s)) {
+          return false;
+        }
+      }
+      return true;
+    };
+    const nearbyUnblockedPlayerSessions = nearbyPlayerSessions.filter(s => notBlocked(s));
+
+    const receivingPlayers = nearbyUnblockedPlayerSessions.map(s => s.player).filter(p => p.id !== sender.id);
     const timestamp = new Date().getTime();
-    nearbyPlayerSessions.forEach(nearbyPlayerSession => {
-      this._listeners.get(nearbyPlayerSession)?.onReceiveChatMessage({
+    nearbyUnblockedPlayerSessions.forEach(nearbyUnblockedPlayerSession => {
+      this._listeners.get(nearbyUnblockedPlayerSession)?.onReceiveChatMessage({
         sender,
         timestamp,
         message: incomingMessage.message,
-        receivingPlayers: nearbyPlayerSession.player.id === sender.id ? receivingPlayers : undefined,
+        receivingPlayers: nearbyUnblockedPlayerSession.player.id === sender.id ? receivingPlayers : undefined,
       });
     });
   }
 
+
+  private onBlockPlayer(session: PlayerSession, playerID: string): void {
+    // create the key, value pair if it is not present in the map
+    if (!this._blockedPlayerSessions.has(session)) {
+      this._blockedPlayerSessions.set(session, []);
+    }
+    // add the blocked player to the block list of 'session'
+    const blockPlayerSession = this._sessions.find((s) => s.player.id === playerID);
+    this._blockedPlayerSessions.get(session)?.push(blockPlayerSession as PlayerSession);
+  }
+
+
+  private onUnblockPlayer(session: PlayerSession, playerID: string): void {
+    const blockedPlayerSession = this._sessions.findIndex((s) => s.player.id === playerID);
```

Comment:
I think this should be:
`this._blockedPlayerSessions.get(session)?.findIndex(...)`

Reply:
I would also rename the variable to something like blockedPlayerSessionIndex so it's clear the variable is an index in the array